code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -31,34 +32,27 @@
 	return FALSE
 
 /obj/item/reagent_containers/food/pill/afterattack(obj/target, mob/user, proximity)
-	if(!proximity)
+	if(!proximity || !target.is_open_container())
+		return
+	else if(!target.reagents.total_volume)
+		to_chat(user, "<span class='warning'>[target] has no liquid to dissolve [src] in.</span>")
 		return
 
-	if(target.is_open_container() != 0 && target.reagents)
-		if(!target.reagents.total_volume)
-			to_chat(user, "<span class='warning'>[target] is empty. Cant dissolve [src].</span>")
-			return
-
-		to_chat(user, "<span class='notify'>You dissolve [src] in [target].</span>")
-		reagents.trans_to(target, reagents.total_volume)
-		for(var/mob/O in viewers(2, user))
-			O.show_message("<span class='warning'>[user] puts something in [target].</span>", 1)
-		spawn(5)
-			qdel(src)
-
-////////////////////////////////////////////////////////////////////////////////
-/// Pills. END
-////////////////////////////////////////////////////////////////////////////////
+	to_chat(user, "<span class='notice'>You dissolve [src] in [target].</span>")
+	for(var/mob/O in oviewers(2, user))
+		O.show_message("<span class='warning'>[user] puts something in [target].</span>", 1)
```

Comment:
Would be great if this were part of `visible_message`. A range of who sees it I mean. Though out of scope for this PR probably